yarn-project/bot/src/runner.ts

```diff
@@ -1,17 +1,18 @@
 import { type PXE, createDebugLogger } from '@aztec/aztec.js';
+import { RunningPromise } from '@aztec/foundation/running-promise';
 
 import { Bot } from './bot.js';
 import { type BotConfig } from './config.js';
 
 export class BotRunner {
   private log = createDebugLogger('aztec:bot');
-  private interval?: NodeJS.Timeout;
   private bot?: Promise<Bot>;
   private pxe?: PXE;
-  private running: Set<Promise<void>> = new Set();
+  private runningPromise: RunningPromise;
 
   public constructor(private config: BotConfig, dependencies: { pxe?: PXE } = {}) {
     this.pxe = dependencies.pxe;
+    this.runningPromise = new RunningPromise(() => this.#safeRun(), config.txIntervalSeconds * 1000);
```

Comment:
Heads up that this changes the semantics of txInterval: before it was the interval between the start of each tx, now it is the interval between a tx is sent (or mined, if the bot is instructed to wait for that) and the next is started.